core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -118,99 +127,91 @@ public static KeyStoreManager getInstance(int tenantId,
      */
     public KeyStore getKeyStore(String keyStoreName) throws Exception {
 
+        if (keyStoreName == null || keyStoreName.isEmpty()) {
+            throw new SecurityException("Key Store Name is null or empty.");
+        }
+
         if (KeyStoreUtil.isPrimaryStore(keyStoreName)) {
             return getPrimaryKeyStore();
         }
 
-        if (isCachedKeyStoreValid(keyStoreName)) {
-            return loadedKeyStores.get(keyStoreName).getKeyStore();
+        if (KeyStoreUtil.isCustomKeyStore(keyStoreName)) {
+            return getCustomKeyStore(keyStoreName);
         }
 
-        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
-        if (registry.resourceExists(path)) {
-            org.wso2.carbon.registry.api.Resource resource = registry.get(path);
-            byte[] bytes = (byte[]) resource.getContent();
-            KeyStore keyStore = KeyStore.getInstance(resource
-                    .getProperty(RegistryResources.SecurityManagement.PROP_TYPE));
-            CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
-            String encryptedPassword = resource
-                    .getProperty(RegistryResources.SecurityManagement.PROP_PASSWORD);
-            String password = new String(cryptoUtil.base64DecodeAndDecrypt(encryptedPassword));
-            ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
-            keyStore.load(stream, password.toCharArray());
-            KeyStoreBean keyStoreBean = new KeyStoreBean(keyStore, resource.getLastModified());
-            resource.discard();
-
-            if (loadedKeyStores.containsKey(keyStoreName)) {
-                loadedKeyStores.replace(keyStoreName, keyStoreBean);
-            } else {
-                loadedKeyStores.put(keyStoreName, keyStoreBean);
-            }
-            return keyStore;
-        } else {
-            throw new SecurityException("Key Store with a name : " + keyStoreName + " does not exist.");
-        }
+        return getTenantKeyStore(keyStoreName);
     }
 
     /**
-     * This method loads the private key of a given key store
+     * This method loads the private key of a given key store.
      *
-     * @param keyStoreName name of the key store
-     * @param alias        alias of the private key
-     * @return private key corresponding to the alias
+     * @param keyStoreName  Name of the key store.
+     * @param alias         Alias of the private key.
+     * @return Private key corresponding to the alias.
+     * @throws Exception    If there is an error when retriving the private key from given keystore.
      */
     public Key getPrivateKey(String keyStoreName, String alias) {
+
+        if (keyStoreName == null || keyStoreName.isEmpty()) {
+            throw new SecurityException("Key Store Name is null or empty.");
+        }
+        if (alias == null || alias.isEmpty()) {
+            throw new SecurityException("Alias is null or empty.");
```

Comment:
```suggestion
            throw new SecurityException("Private key alias is null or empty.");
```

Reply:
Fixed in https://github.com/wso2/carbon-kernel/pull/4025/commits/3aa02c8fd1024563e7aefe9329049af4802dfac3